client/src/App.js

```diff
@@ -1,44 +1,105 @@
-import React, { useState, useEffect } from 'react';
-import logo from './logo.svg';
+import React, { Component } from 'react';
+import { Redirect } from 'react-router';
 import './App.css';
+import { Register } from './components/register/register.component.js';
+import * as EmailValidator from 'email-validator';
 
-function App() {
-  const [message, setMessage] = useState('');
-  useEffect(() => {
-    fetchDataFromServerApi(setMessage);
-  }, []);
+export class App extends Component {
+  constructor(props) {
```

Comment:
Assuming `App.js` just renders register, it shouldn't have any state, since all the logic should be inside register. The rule of thumb for react is each component handles its own state, unless it's shared with other components. In that case, the state should stay in the closest common parent. See [thinking in react](https://reactjs.org/docs/thinking-in-react.html)

State should only contain logic, for style, the basic recommended way is to [use className to style and build it conditionally if needed](https://reactjs.org/docs/faq-styling.html). The 2 popular solutions (but a bit harder to learn) are using a CSS-in-JS library, or CSS modules.

Reply:
I will consider this.